### ISSUE TITLE
Could not find target name 'OSX Player'

### ISSUE THREAD

**Issue by greay** [Unity-Technologies/vscode-unity-debug#104]
Can't attach to Unity standalone build.

## Expected Behavior
I expect it to either automatically connect, or present a list of possible candidates to attach the debugger to.

## Current Behavior
Could not find target name 'OSX Player'. Is it running?

## Steps to Reproduce (for bugs)
1. launch standalone build
2. attempt to attach

## Context
I *can't* attach to the editor, because I'm attempting to debug an issue with IAP. So while this is a development build with script debugging enabled, it also has Mac App Store validation checked.

## Your Environment
* Unity Version: 5.6.4f1
* VS Code Version: 1.25.1
* Version of the debugger tool: 2.6.0
* Operating System and version: 10.13.6

**Comment by greay** [comment]
FWIW, I did a quick test build using Unity 2018.2.0f2 and had the same results.

**Comment by miniwolf** [comment]
Could you see if this new version fix your issue? https://github.com/Unity-Technologies/vscode-unity-debug/releases/download/Version-2.6.4/unity-debug-2.6.4.vsix

**Comment by greay** [comment]
nope. Still "Could not find target name 'OSX Player'. Is it running?"
It won't even show my app as an option in the Unity Process Picker, and I tried two builds – one with MAS validation, and one without.

**Comment by miniwolf** [comment]
Could you try using this build?: https://files.unity3d.com/nicklas/unity-debug-2.6.5.vsix
It has a lot more debug information available that should get posted into the log file.
It is available in `~/.vscode/extensions/unity.unity-debug-2.6.5/bin/UnityDebug-log.txt`

**Comment by miniwolf** [comment]
In addition make sure that your firewall is not blocking the player connection from broadcasting, as it using a UDP connection to the debugger. The debugger then listens on these ports to check if there is any player active.

**Comment by paulevans** [comment]
When launching it does ask if I want to accept connections and I reply yes. :/

Thing is I could have sworn this was working earlier really recently. I think since the very latest Unity that I installed earlier this week it has stopped working? Unity 2018.2.8f1

**Comment by paulevans** [comment]
> @paulevans so are you able to run a different version of unity where it works. But when switching to 18.2.8f1 it does not work? Fx. trying with 2018.1 or the new 18.3 beta build. Does that work?

So between earlier and seeing this I reinstalled 2018.2.8f1 with Visual Studio as I wanted to debug again - though did not completely set it up.

After reading this I immediately installed the newest beta I could see in the hub 2018.03.0b2 and Code *did* did allow me to set breakpoints on a duplicate of the project I was just working on. I have not tried going back to 2018.1

Hope that info helps.

**Comment by paulevans** [comment]
@miniwolf Hmm - on 2018.2.9f1 Visual Studio cannot connect either.

**Comment by miniwolf** [comment]
@paulevans 
Sorry for taking this long to get back to you. That is really interesting information. Going forward I will see if I can reproduce this problem by trying 18.2. If the problem only exists in this revision, that points to some issue inside of Unity.

**Comment by miniwolf** [comment]
Could you have a try with this new version: https://github.com/Unity-Technologies/vscode-unity-debug/releases/download/Version-3.0.1/unity-debug-3.0.1.vsix
There is also a forum post for more general discussions on the new release here: https://forum.unity.com/threads/unity-debugger-extension-for-visual-studio-code-3-x-preview-release.653971/
I will keep you posted.
